build_rules.gradle

```diff
@@ -1260,6 +1261,146 @@ ext.applyGrpcNature = {
 
 /*************************************************************************************************/
 
+ext.applyPortabilityNature = {
+  println "applyPortabilityNature with " + (it ? "$it" : "default configuration") + " for project $project.name"
+  applyJavaNature(failOnWarning: true, enableFindbugs: false, shadowClosure: {
+    // The relocation paths below specifically use the major version number of the dependency
+    // since we assume that packages following semantic versioning rules. For packages
+    // with a major version number that is 0 (implying that the dependency is unstable),
+    // we use the full version identifier in the relocated path (e.g. v0_minor_patch).
+    // For stable dependencies, if two or more packages use the same major version number,
+    // we should bias to vendoring the highest version to minimize jar files instead of
+    // vendoring using their minor or patch version numbers. Some packages may be incompatible
+    // across minor or patch versions and in those instances we should either attempt to update our
+    // usage of the dependency so we don't need to vendor both or choose to vendor
+    // both specifying the minor version in the relocation path.
+
+    // To produce the list of necessary relocations, one needs to start with a set of target
```

Comment:
This also seems like an area to expand into its own document somewhere ("How to update Beam Dependencies"). I wouldn't normally expect to go through this workflow when updating a dependency version. /cc @yifanzou and @chamikaramj, who are adding policy checks to make dependency updates more regular.

Can you mention any tools or commands you're familiar with to capture the set of transitive dependency versions which need to be updated?